docs/annotating_code/supported_types.md

```diff
@@ -0,0 +1,52 @@
+# Supported types
+
+Psalm allows you to specify type information. This is used in a few places:
+
+Psalm assertions:
+```php
+  @psalm-assert <type> $value
+```
+
+Defining the type of a template:
+```php
+/**
+ * @template T
+ */
+class SomeClass { ... }   
+
+/** @var SomeClass<type> $myClass
+$myClass = new SomeClass();
+```
+
+And also constraining templates: 
+```php
+  @template T of <type> 
+```
+
+Where `<type>` can be one of:
+
+| Type          | Template Type | Template constraint | Assertion | Notes |
+|---------------|---------------|---------------------|-----------|-------|
```

Comment:
I'm not sure this table is terribly useful - the takeaway is that value types and bottom types cannot be used as single template constraints - that could probably said in words somewhere.